### PR TITLE
Extensibility: Add activity panel inbox tutorial

### PR DESCRIPTION
**Pull request by allendav**
Adds a tutorial on how extension developers can add their own items to the activity panel inbox.

To test:

- Install docsify if you haven’t already:

	`npm i docsify-cli -g`

- If you needed to do that, you will want to bounce your terminal to get docsify on your path. Then, fire up a local server from the root of the wc-admin plugin.

	`docsify serve docs`

- And then point a browser at http://localhost:3000 to see the newly added Examples content.

### CODE REVIEW

**Review by timmyc — COMMENTED** [review]
Looking good, couple of small edits/ideas to change

File: docs/examples/activity-panel-inbox.md [hunk 1]
```diff
@@ -0,0 +1,239 @@
+Activity Panel Inbox
+====================
+
+Right now, installing and activating WooCommerce and a few extensions quickly results in a cascade of notices vying for the store administrator’s attention (and pushing whatever wp-admin content they were trying to get to on a page down, way down.
```
[the remark this answers is not in the document]
Comment (allendav) [reply docs/examples/activity-panel-inbox.md]: Fixed in https://github.com/woocommerce/wc-admin/pull/753/commits/24f493d21341eedf70c1e2f480bd4fb05d385029

File: docs/examples/activity-panel-inbox.md [hunk 2]
```diff
@@ -0,0 +1,239 @@
...
+
+As part of the Dashboard and Analytics improvements coming to WooCommerce, we’re going to try and do something a bit dramatic to try and clean up this notice-palooza.
```
[the remark this answers is not in the document]
Comment (allendav) [reply docs/examples/activity-panel-inbox.md]: Fixed in https://github.com/woocommerce/wc-admin/pull/753/commits/24f493d21341eedf70c1e2f480bd4fb05d385029

File: docs/examples/activity-panel-inbox.md [hunk 3]
```diff
@@ -0,0 +1,239 @@
...
+
+First, we’re splitting up notices into two broad categories.
+
+## Category 1: Store Alerts
+
+The first category is what we’re calling “Store Alerts” - similar to the notices you have today, this notices are present at the top of administration pages and will look something like this:
```
[the remark this answers is not in the document]
Comment (allendav) [reply docs/examples/activity-panel-inbox.md]: Fixed in https://github.com/woocommerce/wc-admin/pull/753/commits/24f493d21341eedf70c1e2f480bd4fb05d385029

File: docs/examples/activity-panel-inbox.md [hunk 4]
```diff
@@ -0,0 +1,239 @@
...
+
+![Store Alerts](images/activity-panel-store-alerts.png)
+
+But unlike today’s notices, this part of the UI will be reserved for “priority” messages to the administrator - things like version and security updates and critical errors.  What sort of errors are “critical errors”? Well, things like a payment gateway that has stopped working because its connection is down, or when the store is set for physical product support but no shipping methods are defined.
+
+## Category 2: Activity Panel Inbox
+
+The second category is what we’re focusing on in this post - and what we expect the vast majority of extension developers will want to extend - we call it the “Activity Panel Inbox.” It will look something like this:
```
[the remark this answers is not in the document]
Comment (allendav) [reply docs/examples/activity-panel-inbox.md]: Probably not verbatim. Changed to 'example' in https://github.com/woocommerce/wc-admin/pull/753/commits/24f493d21341eedf70c1e2f480bd4fb05d385029

File: docs/examples/activity-panel-inbox.md [hunk 5]
```diff
@@ -0,0 +1,239 @@
...
+
+![Activity Panel Inbox](images/activity-panel-inbox.png)
+
+This section is dedicated to informational content coming from multiple sources such as WooCommerce core, WooCommerce.com Subscription management, extensions activity and store achievements. This section was also thought out to display more insightful content in the future, such as content that provides insights for the store data and that will help with the day to day tasks of managing and optimizing the store performance.
```
[the remark this answers is not in the document]
Comment (allendav) [reply docs/examples/activity-panel-inbox.md]: It was off :) Fixed in https://github.com/woocommerce/wc-admin/pull/753/commits/24f493d21341eedf70c1e2f480bd4fb05d385029

File: docs/examples/activity-panel-inbox.md [hunk 6]
```diff
@@ -0,0 +1,239 @@
...
+
+Each notice or “note” has a type represented by an icon (Gridicon), a title, content, a timestamp and one or two actions (action title + link).
+
+### Inbox Design Constraints and Best Practices
+Extensions can add their own notes via the API we’ll be covering below, but first… some constraints...
```
[the remark this answers is not in the document]
Comment (allendav) [reply docs/examples/activity-panel-inbox.md]: I replaced API with 'data stores' in https://github.com/woocommerce/wc-admin/pull/753/commits/24f493d21341eedf70c1e2f480bd4fb05d385029

File: docs/examples/activity-panel-inbox.md [hunk 7]
```diff
@@ -0,0 +1,239 @@
...
+1. Notes can have 0, 1, or 2 action buttons. Action buttons have URLs. They can either be complete URLs (like in the case of an external link) or partial URLs (like in the case of an admin page on the site) - in the case of partial URLs you’ll want to give a string that is the same as what you’d supply to WordPress core’s admin_url function. You can even include query parameters in that string.
+1. You should store your extension’s name (slug) in the source property of the note. WooCommerce will store ‘woocommerce’ there. For example, the “Panda Payments” extension could use a name like “panda-payments” - this makes it easier to use built in functions to get (or delete) all notes with a certain name.
+1. You can use the name property of a note to store a sub-type of note. For example, if the Panda Payments extension generates notes for both connection problems AND for new features, they might both have ‘panda-payments’ in the source property but then use ‘connection-problem’ and ‘new-feature’ in the name property to distinguish between the two types of notes.
+1. Icons are Gridicons. You can find a gallery here: https://automattic.github.io/gridicons/
```
[the remark this answers is not in the document]
Comment (allendav) [reply docs/examples/activity-panel-inbox.md]: Fixed in https://github.com/woocommerce/wc-admin/pull/753/commits/24f493d21341eedf70c1e2f480bd4fb05d385029

File: docs/examples/activity-panel-inbox.md [hunk 8]
```diff
@@ -0,0 +1,239 @@
...
+
+function wapi_example_two_deactivate() {
+	WooCommerce_Activity_Panel_Inbox_Example_Plugin_Two::remove_activity_panel_inbox_notes();
+}
+register_deactivation_hook( __FILE__, 'wapi_example_two_deactivate' );
+```
+
+## Questions?
+
+This is just the tip of the iceberg for possibilities for your own extensions to WooCommerce. Check out the wc-admin code itself for other examples of working with this fun new feature.
```
[the remark this answers is not in the document]
Comment (allendav) [reply docs/examples/activity-panel-inbox.md]: Fixed in https://github.com/woocommerce/wc-admin/pull/753/commits/24f493d21341eedf70c1e2f480bd4fb05d385029 with links to specific examples

File: docs/examples/activity-panel-inbox.md [hunk 9]
```diff
@@ -0,0 +1,239 @@
...
+1. You can use the name property of a note to store a sub-type of note. For example, if the Panda Payments extension generates notes for both connection problems AND for new features, they might both have ‘panda-payments’ in the source property but then use ‘connection-problem’ and ‘new-feature’ in the name property to distinguish between the two types of notes.
+1. Icons are Gridicons. You can find a gallery here: https://automattic.github.io/gridicons/
+1. As a best practice, have your extension remove its notes on deactivation or at least uninstall.
+
+So, enough rules - let’s get into how to code this up. And surprise, no JavaScript is necessary - you can work with the inbox exclusively through PHP -- the inbox front-end is React based, sure, but it draws all its data from a couple of new database tables and the PHP based REST API (and that’s where you can work with notes).
+
+### Adding and Removing Inbox Items
+
+Here’s a short example plugin that adds a new activity panel inbox note on plugin activation, and removes it on deactivation:
```
[the remark this answers is not in the document]
Comment (allendav) [reply docs/examples/activity-panel-inbox.md]: Hmmm - I kinda like it all embedded here - let me see if there is a way to embed and support easier copy/paste

**Review by timmyc — APPROVED** [review]
Awesomeness!

**Pull request merged**